tests/kolainst/destructive/container-rebase-upgrade

```diff
@@ -29,20 +29,43 @@ set -x
 libtest_prepare_offline
 cd "$(mktemp -d)"
 
+curl -L https://builds.coreos.fedoraproject.org/prod/streams/testing/releases.json > testing.json
+latest_digest=$(jq -r '.releases[-1]["oci-images"][] | select(.architecture == "x86_64") | ."digest-ref"' testing.json | cut -f2 -d@)
+prev_digest=$(jq -r '.releases[-2]["oci-images"][] | select(.architecture == "x86_64") | ."digest-ref"' testing.json | cut -f2 -d@)
+
 image=quay.io/fedora/fedora-coreos
-image_tag=testing-devel
-digest=$(skopeo inspect -n docker://$image:$image_tag | jq -r .Digest)
-image_pull=ostree-remote-registry:fedora:$image@$digest
+image_tag=testing
+image_pull_digest=ostree-remote-registry:fedora:$image@$latest_digest
+image_pull_tag=ostree-remote-registry:fedora:$image:$image_tag
 
 systemctl mask --now zincati
 # Test for https://github.com/ostreedev/ostree/issues/3228
 rpm-ostree kargs --append "foo=\"a b c\""
 rpm-ostree kargs > kargs.txt
 assert_file_has_content_literal kargs.txt "foo=\"a b c\""
-# also test custom origin stuff
-rpm-ostree rebase "${image_pull}" --custom-origin-description "Fedora CoreOS $image_tag stream" --custom-origin-url "$image:$image_tag"
+# also test custom origin stuff and finalization locking
+rpm-ostree rebase "${image_pull_digest}" --custom-origin-description "Fedora CoreOS $image_tag stream" --custom-origin-url "$image:$image_tag" --lock-finalization
 rpm-ostree status > status.txt
 assert_file_has_content_literal status.txt "$image:$image_tag" "Fedora CoreOS $image_tag stream"
+if rpm-ostree finalize-deployment "${prev_digest}" &> out.txt; then
+  cat out.txt
+  assert_not_reached "finalize-deployment with the wrong checksum succeeded!"
+fi
+assert_file_has_content_literal out.txt "Expected staged base checksum $prev_digest, but found $latest_digest"
+
+# test pinned rebase
+ostree admin undeploy 0
+rpm-ostree rebase "${image_pull_tag}" "${prev_digest}"
```

Comment:
This test passed because the brevious rebase targed a full digest refspec : https://github.com/coreos/rpm-ostree/blob/3459988d82870b20f8bc854eb6ed688103ff7050/tests/kolainst/destructive/container-rebase-upgrade#L47

Reply:
Ok reading this some more this is not the actually the reason :)